cwms_radar_api/src/main/java/cwms/radar/data/dao/RatingMetadataDao.java

```diff
@@ -0,0 +1,221 @@
+import cwms.radar.data.dto.rating.TableRatingMetadata;
+import java.sql.Timestamp;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import org.jetbrains.annotations.NotNull;
+import org.jooq.Condition;
+import org.jooq.DSLContext;
+import org.jooq.OrderField;
+import org.jooq.Record;
+import org.jooq.ResultQuery;
+import org.jooq.SelectSeekStepN;
+import org.jooq.conf.ParamType;
+import usace.cwms.db.jooq.codegen.tables.AV_RATING;
+import usace.cwms.db.jooq.codegen.tables.AV_RATING_SPEC;
+
+public class RatingMetadataDao extends JooqDao<RatingSpec> {
+    private static final Logger logger = Logger.getLogger(RatingMetadataDao.class.getName());
+
+    public RatingMetadataDao(DSLContext dsl) {
+        super(dsl);
+    }
+
+    public RatingMetadataList retrieve(String cursor, int pageSize, String office,
+                                       String specIdMask) {
+
+        Object[] parts = CwmsDTOPaginated.decodeCursor(cursor);
+        RatingMetadataList.KeySet keySet = RatingMetadataList.KeySet.build(parts);
+
+        pageSize = RatingMetadataList.KeySet.parsePageSize(parts, pageSize);
+
+        Object[] seekValues = null;
+        if (keySet != null) {
+            seekValues = keySet.getSeekValues();
+        }
+        List<RatingMetadata> ratingMetadata = getRatingMetadata(office, specIdMask, seekValues,
+                pageSize);
+
+        RatingMetadataList.Builder builder = new RatingMetadataList.Builder(pageSize);
+        builder.withKeySet(keySet);
+        builder.withMetadata(ratingMetadata);
+
+        return builder.build();
+    }
+
+    @NotNull
+    public List<RatingMetadata> getRatingMetadata(String office, String specIdMask,
+                                                  Object[] seekValues, int pageSize) {
+        List<RatingMetadata> retval;
+
+        AV_RATING_SPEC specView = AV_RATING_SPEC.AV_RATING_SPEC;
+        AV_RATING ratView = AV_RATING.AV_RATING;
+
+        Condition condition = specView.LOC_ALIAS_CATEGORY.isNull()
+                .and(ratView.LOC_ALIAS_CATEGORY.isNull());
+
+        if (office != null) {
+            condition = condition.and(specView.OFFICE_ID.eq(office));
+        }
+
+        if (specIdMask != null) {
+            condition = condition.and(JooqDao.caseInsensitiveLikeRegex(
+                    specView.RATING_ID, specIdMask));
+        }
+
+
+        Collection<OrderField<?>> fields = new ArrayList<>();
+        fields.add(specView.OFFICE_ID);
+        fields.add(specView.TEMPLATE_ID);
+        fields.add(specView.RATING_ID);
+        fields.add(ratView.OFFICE_ID);
+        fields.add(ratView.RATING_ID);
+        fields.add(ratView.EFFECTIVE_DATE.desc());
+
+
+        SelectSeekStepN<Record> records = dsl.select(specView.RATING_SPEC_CODE,
+                        specView.RATING_ID,
```

Comment:
I have exactly zero idea what it should/could look like but FYI at this length it's not unreasonable to group a few columns into the same line. It would give the read a change to have the column list within the screen at the same time as the rest of the query.

Reply:
good point.